### DISCUSSION TITLE
Dummy class for Ray calls if Ray is not available?

### DISCUSSION THREAD

**Discussion by khromov** [spatie/wordpress-ray#9] [Q&A]
I'm using the Ray WordPress plugin during development, but in production it's not going to be available (The plugin will be distributed through WordPress.org). Therefore I need to figure out a way to have a "dummy" `ray()` method that can be used if the real Ray framework is not available.

Here is a quick solution I came up with. It's supposed to do nothing if Ray is not available and also not throw any errors despite there being various `ray()->...` calls in the codebase. Will this fully cover Rays functionality or did I miss something?

```php
<?php
if(!function_exists('ray')) {
    class Ray_Dummy_Class {
        function __call($funName, $arguments) {
            return new Ray_Dummy_Class();
        }

        static function ray(...$args) {
            return new Ray_Dummy_Class();
        }
        
        function __get($propertyName) {
            return null;
        }

        function __set($property, $value) {
        }
    }

    function ray(...$args) {   
        return Ray_Dummy_Class::ray(...$args);
    }
}
```

**Comment by jeffreyvr** [comment]
If you install the plugin through wordpress.org, even when in production, it will still make the `ray` function available. It might not work like it does locally, but the function will exist. So I don't think this dummy class is needed. Or am I missing something maybe?

**Reply by khromov** [reply]
@jeffreyvr What I meant was building a plugin and then distributing it through WordPress.org. Users who download my plugin won't have Ray installed, nor should they be required to. It would also be a fatal error for users to activate my plugin without having Ray installed which is a no-go.

**Comment by Mamaduka** [comment]
You might want to stub the [`rd`](https://spatie.be/docs/ray/v1/usage/framework-agnostic-php-project#halting-the-php-process) function as well.  You can also return an instance of the class in the call method instead of creating a new one.

Otherwise, it looks good.
